crates/polars-plan/src/plans/aexpr/function_expr/schema.rs

```diff
@@ -276,9 +284,17 @@ impl IRFunctionExpr {
                     return Ok(dtype);
                 }
 
-                let num_infers = dims.iter().filter(|d| matches!(d, ReshapeDimension::Infer)).count();
+                if dims[1..]
+                    .iter()
+                    .any(|d| matches!(d, ReshapeDimension::Infer))
+                {
+                    polars_bail!(InvalidOperation: "can only infer the first dimension");
+                }
 
-                polars_ensure!(num_infers <= 1, InvalidOperation: "can only specify one inferred dimension");
+                let num_infers = dims
```

Comment:
I think this can now be `usize::from(matches!(dims[0], ReshapeDimension::Infer))`. But, in general, if we can just have a boolean here now saying `infers_dimension` as there is now only 1 dimension to infer.